src/Entity/PropertyMapper.php

```diff
@@ -337,6 +351,123 @@ protected function checkAnding(array &$context, array $mergedArray = []) {
     }
   }
 
+  /**
+   * Converts the final value to a type if it's been set.
+   *
+   * Example YAML:
+   * @code
+   * field_map:
+   *   typeCastExample: !!string field_movie_primary_genre.id
+   * @endcode
+   *
+   * The end structure should return:
+   * @code
+   * "typeCastExample": "1234"
+   * @endcode
+   *
+   *
+   * @param string $dest_prop
+   *  The final destination property.
+   * @param $final
+   *  The final_source_prop_value from yamlPropertyMapper().
+   *
+   * @return mixed
+   */
+  protected function setType($dest_prop, $final) {
+    $value = $this->getValue($dest_prop);
+    if(isset($value['data_type']) && isset($value['data'])) {
+      settype($final, $value['data_type']);
+    }
+    return $final;
+  }
+
+  /**
+   * Value getter for stored values.
+   *
+   * @param string $dest_prop
+   *  The final destination property
+   *
+   * @return mixed
+   */
+  protected function getValue($dest_prop = NULL) {
+    if (!empty($this->value)) {
+      if ($dest_prop && isset($this->value[$dest_prop])) {
+        return $this->value[$dest_prop];
+      }
+    }
+  }
+
+  /**
+   * Checks and sets any typecasting. This will override any final values.
+   *
+   * @param string $source_prop
+   * @param string $dest_prop
+   */
+  protected function checkTypeCasting(&$source_prop, $dest_prop){
+    preg_match("/!!\b([a-z]*){2,}\b/", $source_prop, $matches);
+    if (count($matches) > 0) {
+      $type = str_replace('!!', '', $matches[0]);
```

Comment:
@asauterChicago would you also add example type casting on this function, similar to what's in the comment block of `checkOring()`?